presto-spi/src/test/java/com/facebook/presto/spi/block/TestBlockFlattenner.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Optional;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.SmallintType.SMALLINT;
+import static com.facebook.presto.spi.type.TinyintType.TINYINT;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertSame;
+import static org.testng.Assert.assertTrue;
+
+public class TestBlockFlattenner
+{
+    private ArrayAllocator allocator;
+    private BlockFlattener flattener;
+
+    @BeforeClass
+    public void setup()
+    {
+        this.allocator = new SimpleArrayAllocator();
+        this.flattener = new BlockFlattener(allocator);
+    }
+
+    @Test
+    public void testLongArrayIdentityDecode()
+    {
+        Block block = createLongArrayBlock(1, 2, 3, 4);
+        try (BlockLease blockLease = flattener.flatten(block)) {
+            Block flattenedBlock = blockLease.get();
+            assertSame(flattenedBlock, block);
+        }
+    }
+
+    @Test
+    public void testNestedDictionaryRLELongArray()
```

Comment:
I don't see a test for nested dictionary/dictionary combination. Also it would be good to add 3 levels of nested blocks because that's when the release of leasedMap is tested.